test/CountedPtrBenchmarks.h

```diff
@@ -0,0 +1,282 @@
+*/
+/* The original authors have included a copy of the license specified above in the
+   'Docs' folder. See 'gpl.txt'
+*/
+/* We welcome the use of the Mezzanine engine to anyone, including companies who wish to
+   Build professional software and charge for their product.
+
+   However there are some practical restrictions, so if your project involves
+   any of the following you should contact us and we will try to work something
+   out:
+    - DRM or Copy Protection of any kind(except Copyrights)
+    - Software Patents You Do Not Wish to Freely License
+    - Any Kind of Linking to Non-GPL licensed Works
+    - Are Currently In Violation of Another Copyright Holder's GPL License
+    - If You want to change our code and not add a few hundred MB of stuff to
+        your distribution
+
+   These and other limitations could cause serious legal problems if you ignore
+   them, so it is best to simply contact us or the Free Software Foundation, if
+   you have any questions.
+
+   Joseph Toppi - toppij@gmail.com
+   John Blackwood - makoenergy02@gmail.com
+*/
+#ifndef Mezz_Foundation_CountedPtrBenchmarks_h
+#define Mezz_Foundation_CountedPtrBenchmarks_h
+
+/// @file
+/// @brief This contains benchmarks tests on the performance of the CountedPtr class.
+
+#include "MezzTest.h"
+
+#include "CountedPtr.h"
+#include "CountedPtrTests.h"
+
+BENCHMARK_TEST_GROUP(CountedPtrBenchmarks,CountedPtrBenchmarks)
+{
+    using namespace Mezzanine;
+    using namespace CountedPtrTesting;
+
+    Integer OutputE = 0;
+    Integer OutputI = 0;
+    Integer OutputS = 0;
+
+    Boole destructFlag = false; // Only used here for construction of the test classes.
+
+    using ClockType = std::chrono::high_resolution_clock;
+    using TimeType = ClockType::time_point;
+    using DurationType = ClockType::duration;
+
+    TimeType Begin;
+    TimeType End;
+
+    {
+        TestLog << "\n" << "CountedPtr Benchmarks" << "\n\n"
+                << "The objects being created all change a variable on destruction and have initializing, "
+                << "but otherwise trivial constructors. This is useful only for comparing the speeds of the point "
+                << "constructs on this platform, not for providing objective pointer dereferencing costs." << "\n\n";
+        const Whole TestCount=10000000;
+
+        /////////////////////////////////////
+        // Create tests
+
+        /// Raw Pointers
+        Begin = ClockType::now();
+        for( Whole Count = 0 ; Count < TestCount ; Count++ )
+        {
+            FooExternal* PtrR = new FooExternal(&destructFlag, 1);
+            OutputS = PtrR->Value;
+            delete PtrR;
+        }
+        End = ClockType::now();
+        DurationType RawPtrTime = End - Begin;
+        TestLog << OutputS << " - Creating and Dereferencing a raw pointer " << TestCount
+                << " times without reference counting took: " << RawPtrTime.count() << " Microseconds."
```

Comment:
You can use Mezzanine::Testing::PrettyDurationString(Duration) to get a formatted string like what the unit tests emit in their summary.